sdk/storage/storage-blob/src/BlobClient.ts

```diff
@@ -1119,6 +1269,151 @@ export class BlobClient extends StorageClient {
     });
   }
 
+  /**
+   * Set the owner, owner group or access control list for a directory.
+   *
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update
+   *
+   * @param {string} accessControl Sets POSIX access control rights on files and directories. The value is a comma-separated list
+   *                               of access control entries. Each access control entry (ACE) consists of a scope, a type, a user
+   *                               or group identifier, and permissions in the format "[scope:][type]:[id]:[permissions]".
+   *
+   * @param {string} accessControl
+   * @param {BlobSetAccessControlOptions} [options={}]
+   * @returns {Promise<Models.BlobSetAccessControlResponse>}
+   * @memberof BlobClient
+   */
+  public async setAccessControl(
+    accessControl: string,
+    options: BlobSetAccessControlOptions = {}
+  ): Promise<Models.BlobSetAccessControlResponse> {
+    options.blobAccessConditions = options.blobAccessConditions || {};
+    const aborter = new ContextAborter(options.abortSignal || AbortSignal.none, DfsContext);
+    return this.blobContext.setAccessControl({
+      abortSignal: aborter,
+      owner: options.owner,
+      group: options.group,
+      posixAcl: accessControl,
+      leaseAccessConditions: options.blobAccessConditions.leaseAccessConditions,
+      modifiedAccessConditions: options.blobAccessConditions.modifiedAccessConditions
+    });
+  }
+
+  /**
+   * Set the owner, owner group and their permissions for a blob.
+   *
+   * @see https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/update
+   *
+   * @param {string} [permissions] Optional only when `options.owner` or `options.group` is provided.
+   *                               Valid if Hierarchical Namespace is enabled for the account. Sets POSIX
+   *                               access permissions for the file owner, the file owning group, and others. Each class may be
+   *                               granted read, write, or execute permission.  The sticky bit is also supported.  Both symbolic
+   *                               (rwxrw-rw-) and 4-digit octal notation (e.g. 0766) are supported.
+   * @param {string} [permissions]
+   * @param {BlobSetPermissionsOptions} [options={}]
+   * @returns {Promise<Models.BlobSetAccessControlResponse>}
+   * @memberof BlobClient
+   */
+  public async setPermissions(
+    permissions?: string,
+    options: BlobSetPermissionsOptions = {}
+  ): Promise<Models.BlobSetAccessControlResponse> {
+    if (!permissions && !options.owner && !options.group) {
+      throw TypeError(
+        "options.owner or options.group must be provided when permissions parameter is empty."
```

Comment:
Minor: Lower case o or Options?

Reply:
`options` refers to the `options` parameter, so using lower case.

Reply:
options refers to the options parameter, so using lower case.

---
In reply to: [318944504](https://github.com/Azure/azure-sdk-for-js/pull/4874#discussion_r318944504) [](ancestors = 318944504)